pi5/hw_pwm.go

```diff
@@ -166,6 +173,35 @@ func (pwm *pwmDevice) callGPIOReadall() {
 	}
 }
 
+/*
+Each pwm line corresponds to a GPIO Pin. The pi5 mapping is:
+
+	Line 0 -> GPIO 12
+	Line 1 -> GPIO 13
+	Line 2 -> GPIO 18
+	Line 3 -> GPIO 19
+
+Other mappings for different pis can be found here:  https://pypi.org/project/rpi-hardware-pwm/#modal-close
+
+Use the board.setPin to set the mode to PWM or GPIO using this helper method. Pin Mode will either be 'HPWM' or 'GPIO'
+*/
+func (pwm *pwmDevice) SetPinMode(pinMode byte) (err error) {
+	switch pwm.line {
+	case 0:
+		err = pwm.board.setPin(12, pinMode)
+	case 1:
+		err = pwm.board.setPin(13, pinMode)
+	case 2:
+		err = pwm.board.setPin(18, pinMode)
+	case 3:
+		err = pwm.board.setPin(19, pinMode)
+	default:
+		return errors.New("attemping to set PWM pin mode but pwm line is not valid")
```

Comment:
The purpose of this error was to explicitly prevent setting the mode of pins that do not support PWM. I guess that wouldn't actually happen, because all of these methods are used with a PWM device.

Reply:
Right now, I don't think we want to perform pin control for any other purpose, so I restricted it to just pins that support PWM.